addons/web/static/src/js/views/list/list_controller.js

```diff
@@ -888,6 +888,13 @@ var ListController = BasicController.extend({
             this._enableButtons();
         }
     },
+    /**
+     * @override
+     * @private
+     */
+    _onOdooLeave: function () {
+        this.renderer.unselectRow();
```

Comment:
You take the usual path here, so you'll wait for onchanges, and thus the record might not be saved. Am I missing something?

To reproduce: have a list view with two fields A and B, and an onchange on A which changes B, set A, close the tab.